app/services/cosmos/cosmos.types.ts

```diff
@@ -58,32 +60,24 @@ export interface CosmosTxQueryResult {
   timestamp: string
 }
 
-export interface CosmosSendResult {
-  hash: string
-  sequence: any
-  included: () => Promise<CosmosTxQueryResult>
+export interface CosmosMessage {
+  msgs: {
+    typeUrl: string
+    value: any
+  }[]
+  signerAddress: string
 }
 
-export interface CosmosSignature {
-  signature: Buffer
-  publicKey: Buffer
+export interface CosmosMessageToSign extends CosmosMessage {
+  fee: {
+    amount: CosmosCoinResult[]
+    gas: string
+  }
+  memo?: string
 }
 
-export interface CosmosMessage {
-  message: any
-  simulate: ({ memo }: {
-    memo?: any
-  }) => Promise<number>
-  send: (
-    meta: {
-      gas: any
-      gasPrices?: any
-      memo?: any
-    },
-    signer: (
-      signMessage: string,
-    ) => CosmosSignature
-  ) => Promise<CosmosSendResult>
+export interface CosmosSigner {
```

Comment:
For overall context, I prefer `CosmosSigningClient` over `CosmosSigner`

Reply:
RIght, renamed